tests/engine/processes/workchains/test_restart.py

```diff
@@ -101,6 +105,10 @@ def test_unhandled_reset_after_handled(generate_work_chain, generate_calculation
     # Exit status 400 of the last calculation job will be handled and so should reset the flag
     process.ctx.children.append(generate_calculation_node(exit_status=400))
     result = process.inspect_process()
+
+    # Even though `handler_a` was followed by `handler_b`, we should retrieve the exit_code from `handler_a` because
+    # `handler_b` returned `None` which should not overwrite the last report.
     assert isinstance(result, ExitCode)
-    assert result.status == 0
+    assert result.status == 418
+    assert result.message == 'IMATEAPOT'
     assert process.ctx.unhandled_failure is False
```

Comment:
If we are testing the reset, and although it might be redundant because you already check that a calculation with an exit status != 0 sets off the `process.ctx.unhandled_failure`, shouldn't it be better to check that the status actually change and was set back? (so perhaps `assert process.ctx.unhandled_failure is True` just before calling `result = process.inspect_process()`)